devito/arch/archinfo.py

```diff
@@ -376,6 +377,35 @@ def get_cuda_path():
     return None
 
 
+@memoized_func
+def get_m1_llvm_path():
+    # Check if Apple's llvm is installed (installable via Homebrew), which supports
+    # OpenMP.
+    # Check that we are on apple system
+    if sys.platform != 'darwin':
+        raise ValueError('Apple LLVM is only available on Mac OS X.')
+    # *** First check if LLVM is installed
+    ver = run(["clang", "--version"], stdout=PIPE, stderr=DEVNULL).stdout.decode("utf-8")
+    # *** Second check if this clang version targets arm64 (M1)
+    if "arm64-apple" in ver:
+        # *** Third extract install path. clang version command contains the line:
+        # InstalledDir: /path/to/llvm/bin
+        # which points to the llvm root directory we need for libraries and includes
+        prefix = [v.split(': ')[-1] for v in ver.split('\n')
+                  if "InstalledDir" in v][0][:-4]
+        # ** Fourth check if the libraries are installed
+        if os.path.exists(os.path.join(prefix, 'lib', 'libomp.dylib')):
+            libs = "%s/lib" % prefix
```

Comment:
`os.path.join(prefix, lib)` a bit more portable, but hey, nitpicking here...

Reply:
You mean like the line above 😅 will fix

Reply:
fixed